core/templates/dev/head/pages/exploration_editor/RouterServices.js

```diff
@@ -44,14 +44,20 @@ oppia.factory('routerService', [
       return $location.path();
     }, function(newPath, oldPath) {
       if (newPath === '') {
-        $location.path(oldPath);
+        if (oldPath === '/') {
+          $window.history.go(-1);
+        } else {
+          $location.path(oldPath);
+        }
         return;
       }
-
       if (!oldPath) {
         // This can happen when clicking on links whose href is "#".
         return;
       }
+      if (newPath === '/' && oldPath.includes('/gui/')) {
```

Comment:
The context I'm referring to here is that there is interesting redirection behaviour happening when the user comes from the dashboard, but not when the user comes from other pages. So, this seems like the wrong place to deal with that redirection problem; that ought to be something that happens on the dashboard page itself.

There is a part of your comment that I do not understand. If I am on the library page and I copy/paste manually a link to the /create page for an exploration which I own, why would pressing the back button enter a redirection loop?